### PR TITLE
Feature/#411 - Raise error if config_id is an entity attribute

### PR DESCRIPTION
**Pull request by trgiangdo**
https://github.com/Avaiga/taipy/issues/411

### CODE REVIEW

File: taipy/core/data/data_node.py [hunk 1]
```diff
@@ -120,6 +121,22 @@ def __init__(
...
+        self._config_id = val
+
+    @property
+    def owner_id(self):
+        return self._owner_id
+
+    @owner_id.setter
+    def owner_id(self, val):
+        self._owner_id = val
+
```
Comment (jrobinAV) [inline taipy/core/data/data_node.py]: I believe it was on purpose not to have the properties.

Both `config_id` and owner_id must not be editable. You cannot change the config_id of a data node at runtime.

File: taipy/core/scenario/scenario.py [hunk 2]
```diff
@@ -156,6 +155,14 @@ def __getattr__(self, attribute_name):
...
         raise AttributeError(f"{attribute_name} is not an attribute of scenario {self.id}")
 
+    @property
+    def config_id(self):
+        return self._config_id
+
+    @config_id.setter
+    def config_id(self, val):
+        self._config_id = val
+
```
Comment (jrobinAV) [inline taipy/core/scenario/scenario.py]: It must not be editable.

File: taipy/core/task/task.py [hunk 3]
```diff
@@ -109,6 +107,22 @@ def properties(self):
...
+        self._config_id = val
+
+    @property
+    def owner_id(self):
+        return self._owner_id
+
+    @owner_id.setter
+    def owner_id(self, val):
+        self._owner_id = val
+
```
Comment (jrobinAV) [inline taipy/core/task/task.py]: Same here. They must not be editable.

**Review by jrobinAV — CHANGES_REQUESTED** [review]
I believe this PR is promising but does not cover the use cases. It prevents a `data_node_config` from being named like a data node attribute, a `scenario_config` from being named like a scenario attribute, etc. 
But actually we want to prevent a `data_node_config` from having the name of a scenario attribute.

1. For instance, a data_node_config cannot be named "tags" because we want to be able to do 'scenario.<data_node_config_name>.read()' and if <data_node_config_name> is "tags", there is a conflict with the 'tags' property.
2. Another question, why is it a pb to name a scenario "tags", or a data_node "skippable". I cannot see any conflict.

File: taipy/core/data/data_node.py (hunk 1 above)
Comment (trgiangdo) [reply taipy/core/data/data_node.py]: It was editable before since the `config_id` was public. To avoid it to be editable, we can remove the setter

Comment (jrobinAV) [reply taipy/core/data/data_node.py]: yes, indeed.

**Comment by trgiangdo** [comment]
In the latest commits:
- The id of a `DataNodeConfig` should not be named as a Task or Scenario attribute
- The id of a `TaskConfig` should not be named as a Scenario attribute

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**